src/aggregate/aggregate_exec.c

```diff
@@ -478,6 +478,10 @@ static void sendChunk_Resp2(AREQ *req, RedisModule_Reply *reply, size_t limit,
       QueryWarningsGlobalStats_UpdateWarning(QUERY_WARNING_CODE_OUT_OF_MEMORY_COORD, 1, !IsInternal(req));
     }
 
```

Comment:
consider first to check if we have qctx->err, then check its type to avoidmultiple branching in sunny day scenario